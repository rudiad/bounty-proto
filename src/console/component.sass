.console
  bottom: 0
  position: absolute
  border: 1px solid black
  padding: .7em
  left: 0
  right: 0
  margin-left: auto
  margin-right: auto
  height: 10em
  color: white
  background-color: #241d30
  opacity: 0.8
  overflow-y: auto
  font-family: "Lucida Console", Monaco, monospace
  z-index: -1
