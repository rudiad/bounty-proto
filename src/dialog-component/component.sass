$top-bar-dimension: 40px

.dialog__backdrop
  position: fixed
  top: 0
  bottom: 0
  left: 0
  right: 0
  background-color: #19171C
  opacity: 0.75
  padding: 50
  height: 100vh

.dialog
  background-color: white
  border-radius: 5
  width: 66%
  margin: 0 auto
  border: 1px solid #DAD8DE
  position: absolute
  top: 20%
  left: 20%

.container
  display: flex

.container__block
  display: inline-block

.container__left-block
  @extend .container__block
  float: left
  margin-left: 20px

.container__right-block
  @extend .container__block
  float: right

.top-bar
  @extend .container
  height: $top-bar-dimension
  justify-content: space-between

.top-bar-element
  width: auto
  text-align: center
  vertical-align: middle
  line-height: $top-bar-dimension
  font-weight: lighter

.top-bar-element-left
  @extend .container__left-block
  @extend .top-bar-element

.top-bar-element-right
  @extend .container__right-block
  @extend .top-bar-element

.dialog__divider-width
  width: 96%

@mixin bottom-bar-box($color, $bg-color)
  color: $color
  background-color: $bg-color
  border: 1px solid #7D55C7
  float: left
  text-align: center
  width: 80px
  height: 30px
  vertical-align: middle
  line-height: 30px
  cursor: pointer
  font-size: 16px
  margin-left: 20px
  margin-top: 10px
  margin-bottom: 30px

.title
  font-size: 13px
  font-weight: lighter
