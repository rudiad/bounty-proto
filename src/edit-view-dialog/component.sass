@import "../dialog-component/component.sass"

.edit-view__background
  @extend .dialog__backdrop

.edit-view__dialog-component
  @extend .dialog
  height: 275px
  width: 275px
  left: 40%

.edit-view__dialog-mobile
  @extend .dialog
  height: 275px
  width: 415px
  left: 35%

.edit-view__dialog-overlay
  @extend .dialog
  height: 275px
  width: 275px
  left: 40%

.edit-view__content
  @extend .container__left-block
  overflow-y: auto
  height: 50%
  width: 95%
  display: flex
  flex-direction: column

.edit-subcontainer__input
  margin-top: .5em
  margin-bottom: .5em

.dialog__top-bar-container
  @extend .top-bar

.top-bar-container__title
  @extend .top-bar-element-left
  margin-top: 5px

.top-bar-container__escape
  @extend .top-bar-element-right
  width: $top-bar-dimension
  cursor: pointer
  width: 4em
  margin-top: 5px

.dialog__divider
  @extend .dialog__divider-width

.dialog__bottom-bar
  @extend .container__left-block

.bottom-bar__save
  @include bottom-bar-box(white, #7D55C7)

.bottom-bar__cancel
  @include bottom-bar-box(#7D55C7, white)
  margin-left: 10px
