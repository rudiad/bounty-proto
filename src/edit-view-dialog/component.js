import React, { Component } from 'react';
import PropTypes from 'prop-types';
import closeButton from '../img/close_icon.png';
import './component.sass';

export class EditViewDialog extends Component {
  constructor(args) {
    super(args);
    this.state = {
      x: 0,
      y: 0,
    }
  }

  onChange = (input) => {
    const newState = {};
    newState[input.target.name] = input.target.value;
    this.setState(newState);
  }

  componentDidMount() {
    this.props.views.forEach(element => {
      if (element.id === this.props.idToEdit) {
        this.setState({
          x: element.x,
          y: element.y,
        });
      }
    });
  }

  render() {
    if (!this.props.show) {
      return null;
    }

    return (
      <div className="edit-view">
        <div className="edit-view__background"/>
        <div className="edit-view__dialog">
          <div className="dialog__top-bar-container">
            <div className="top-bar-container__title"> Edit View </div>
            <div className="top-bar-container__escape" onClick={this.props.closeHandler}><img alt="Close" src={closeButton}/></div>
          </div>
          <hr className="dialog__divider"/>
          <div className="edit-view__content">
            <div className="type-and-size-container__type-title">
                  Position (%)
            </div>
              <div className='overlay-custom-container'>
                <div className='overlay-custom-container'>
                <div className="edit-subcontainer__input">
                  <label className="inputs__option-label">X</label>
                  <input type="text" name="x" value={this.state.x} onChange={this.onChange}/>
                </div>
                <div className="edit-subcontainer__input">
                  <label className="inputs__option-label">Y</label>
                  <input type="text" name="y" value={this.state.y} onChange={this.onChange}/>
                </div>
                </div>
              </div>
            </div>
          <hr className="dialog__divider"/>
          <div className="dialog_bottom-bar">
            <div className="bottom-bar__save" onClick={() => this.props.saveViewHandler({ x: this.state.x, y: this.state.y })}> Save </div>
            <div className="bottom-bar__cancel" onClick={this.props.closeHandler}> Cancel </div>
          </div>
        </div>
      </div>
    );
  }
}

EditViewDialog.propTypes = {
  show: PropTypes.bool.isRequired,
  idToEdit: PropTypes.string.isRequired,
  views: PropTypes.array.isRequired,
  closeHandler: PropTypes.func.isRequired,
  saveViewHandler: PropTypes.func.isRequired,
};
