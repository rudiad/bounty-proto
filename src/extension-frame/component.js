import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

const { ExtensionPlatform, ExtensionViewType} = window['extension-coordinator'];

const IFRAME_CLASS = 'extension-frame';
const EXTENSION_FRAME_INIT_ACTION = 'extension-frame-init';

export class ExtensionFrame extends Component {
  componentDidMount() {
    if (this.iframe) {
      this.iframe.onload = this._extensionFrameInit;
    }
  }

  render() {
    return (
      <iframe
      ref={this._bindIframeRef}
        src={process.env.PUBLIC_URL + '/extension-frame.html'}
        frameBorder={0}
        className={'rig-frame ' + IFRAME_CLASS}
        title={this.props.frameId}/>
    );
  }

  _bindIframeRef = (iframe) => {
    this.iframe = iframe;
  }

  _extensionFrameInit = () => {
    const extension = {
      anchor: this.props.type,
      channelId: this.props.extension.channelId,
      loginId: null,
      extension: this.props.extension,
      mode: this.props.mode,
      platform: (this.props.type === ExtensionViewType.Mobile) ? ExtensionPlatform.Mobile : ExtensionPlatform.Web,
      trackingProperties: {},
      iframeClassName: IFRAME_CLASS,
    }
    const data = {
      extension: extension,
      action: EXTENSION_FRAME_INIT_ACTION,
      frameId: this.props.frameId,
    }
    this.iframe.contentWindow.postMessage(data, '*');
  }

  _onIdentityLinked(isLinked) {}

  _onFrameDoubleClick(evt) {
    evt.preventDefault();
  }

  _onModalRequested(confirmationRequest) {}
}

ExtensionFrame.propTypes = {
  className: PropTypes.string.isRequired,
  frameId: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
};
