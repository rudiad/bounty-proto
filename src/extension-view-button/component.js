import React, { Component } from 'react';
import PropTypes from 'prop-types';
import line from '../img/rig_line.png';
import plus from '../img/plus_icon.png';
import './component.sass'

export class ExtensionViewButton extends Component {
  render() {
    return (
      <div className='button-wrapper'>
        <img alt='line' className='background' src={line} />
        <div className="circle-button" onClick={this.props.onClick}>
          <img alt='Add new view' src={plus} />
        </div>
      </div>
    )
  }
}

ExtensionViewButton.propTypes = {
  onClick: PropTypes.func.isRequired
}
