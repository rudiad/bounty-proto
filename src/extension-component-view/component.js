import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ExtensionFrame } from '../extension-frame';
const { ExtensionMode, ExtensionViewType, getComponentPositionFromView, getComponentSizeFromView } = window['extension-coordinator'];

export class ExtensionComponentView extends Component {
  computeViewStyles() {
    const extension = this.props.extension;
    const positionFromView = getComponentPositionFromView(
      this.props.overlaySize.width,
      this.props.overlaySize.height,
      {
        x: this.props.position.x * 100,
        y: this.props.position.y * 100,
      });
    const sizeFromView = getComponentSizeFromView(
      this.props.overlaySize.width,
      this.props.overlaySize.height,
      extension.views.component);

    let viewStyles = {
      border: '1px solid #7D55C7',
      position: 'absolute',
      left: positionFromView.x + 'px',
      top: positionFromView.y + 'px',
      width: `${sizeFromView.width}px`,
      height: `${sizeFromView.height}px`,
    }

    if (extension.views.component.zoom) {
      viewStyles = {
        width: `${sizeFromView.width / sizeFromView.zoomScale}px`,
        height: `${sizeFromView.height / sizeFromView.zoomScale}px`,
        transformOrigin: '0 0',
        transform: `scale(${sizeFromView.zoomScale})`,
      }
    }

    return viewStyles;
  }

  render() {
    return (
      <div
        className="view component-view"
        style={{
          width: this.props.overlaySize.width + 'px',
          height: this.props.overlaySize.height + 'px',
        }}>
          <div style={this.computeViewStyles()}>
          <ExtensionFrame
            className="view"
            frameId={`frameid-${this.props.id}`}
            extension={this.props.extension}
            type={ExtensionViewType.Component}
            mode={ExtensionMode.Viewer}
          />
        </div>
      </div>
    );
  }
}

ExtensionComponentView.propTypes = {
  id: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  overlaySize: PropTypes.object.isRequired,
  position: PropTypes.object.isRequired,
  role: PropTypes.string,
};
