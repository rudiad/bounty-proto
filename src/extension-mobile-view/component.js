import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ExtensionFrame } from '../extension-frame';
import { MobileOrientation } from '../constants/mobile';
const { ExtensionMode, ExtensionViewType} = window['extension-coordinator'];

export class ExtensionMobileView extends Component {
  computeFrameStyles() {
    let frameStyles;

    if (this.props.orientation === MobileOrientation.Portrait) {
      const height = Math.floor(this.props.frameSize.height * 0.65);
      frameStyles = {
        width: `${this.props.frameSize.width}px`,
        height: `${height}px`,
        bottom: 0
      }
    } else {
      const width = Math.floor(this.props.frameSize.height * 0.28);
      frameStyles = {
        width: `${width}px`,
        height: `${this.props.frameSize.width}px`,
        right: 0
      }
    }

    frameStyles.position = 'absolute';
    return frameStyles;
  }

  computeViewStyles() {
    let viewStyles;
    if (this.props.orientation === MobileOrientation.Portrait) {
      viewStyles = {
        width: this.props.frameSize.width + 'px',
        height: this.props.frameSize.height + 'px',
      }
    } else {
      viewStyles = {
        width: this.props.frameSize.height + 'px',
        height: this.props.frameSize.width + 'px',
      }
    }

    viewStyles.background = '#322F37';
    return viewStyles;
  }
  render() {
    return (
      <div
        className="view component-view"
        style={this.computeViewStyles()}>
          <div style={this.computeFrameStyles()}>
          <ExtensionFrame
            className="view"
            frameId={`frameid-${this.props.id}`}
            extension={this.props.extension}
            type={ExtensionViewType.Mobile}
            mode={ExtensionMode.Viewer}
          />
        </div>
      </div>
    );
  }
}

ExtensionMobileView.propTypes = {
  id: PropTypes.string.isRequired,
  orientation: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  frameSize: PropTypes.object.isRequired,
  position: PropTypes.object.isRequired,
  role: PropTypes.string,
};
