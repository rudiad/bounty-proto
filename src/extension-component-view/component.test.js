import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { ExtensionComponentView } from './component';
import { ViewerTypes } from '../constants/viewer-types';
import { ExtensionForTest } from '../tests/constants/extension';

describe('<ExtensionComponentView />', () => {
  const setupShallow = setupShallowTest(ExtensionComponentView, () => ({
    id: '1',
    extension: ExtensionForTest,
    frameSize: { width: 800, height: 600 },
    position: { x: 10, y: 10 },
    role: ViewerTypes.Broadcaster,
  }));

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when zoom is true', () => {
    const ext = Object.assign({}, ExtensionForTest, {
      ...ExtensionForTest,
      views: {
        component: {
          aspectWidth: 3000,
          aspectHeight: 2000,
          zoom: true,
        },
        config: {
          viewerUrl: "test",
        },
        liveConfig: {
          viewerUrl: "test",
        },
        panel: {
          viewerUrl: 'test',
          height: '300px'
        }
      },
    });
    const { wrapper } = setupShallow({
      extension: ext,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
