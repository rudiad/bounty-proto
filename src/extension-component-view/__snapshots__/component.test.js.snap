// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionComponentView /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionComponentView
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    id="1"
    overlaySize={
        Object {
            "height": 600,
            "width": 800,
          }
    }
    position={
        Object {
            "x": 10,
            "y": 10,
          }
    }
    role="Broadcaster"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
                Object {
                        "border": "1px solid #7D55C7",
                        "height": "10px",
                        "left": "20px",
                        "position": "absolute",
                        "top": "20px",
                        "width": "10px",
                      }
        }
>
        <ExtensionFrame
                className="view"
                extension={
                        Object {
                                "authorName": "test",
                                "channelId": "channelId",
                                "description": "description",
                                "iconUrl": "icon_url",
                                "id": "id",
                                "name": "name",
                                "requestIdentity": false,
                                "sku": "sku",
                                "state": "state",
                                "summary": "summary",
                                "token": "token",
                                "vendorCode": "vendorCode",
                                "version": "0.1",
                                "views": Object {
                                  "component": Object {
                                    "aspectHeight": 3000,
                                    "aspectWidth": 2500,
                                    "viewerUrl": "test",
                                    "zoom": false,
                                  },
                                  "config": Object {
                                    "viewerUrl": "test",
                                  },
                                  "liveConfig": Object {
                                    "viewerUrl": "test",
                                  },
                                  "panel": Object {
                                    "viewerUrl": "test",
                                  },
                                },
                                "whitelistedConfigUrls": Array [
                                  "foo",
                                ],
                                "whitelistedPanelUrls": Array [
                                  "bar",
                                ],
                              }
                }
                frameId="frameid-1"
                mode="viewer"
                type="component"
        />
</div>,
      "className": "view component-view",
      "style": Object {
        "height": "600px",
        "width": "800px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ExtensionFrame
          className="view"
          extension={
                    Object {
                              "authorName": "test",
                              "channelId": "channelId",
                              "description": "description",
                              "iconUrl": "icon_url",
                              "id": "id",
                              "name": "name",
                              "requestIdentity": false,
                              "sku": "sku",
                              "state": "state",
                              "summary": "summary",
                              "token": "token",
                              "vendorCode": "vendorCode",
                              "version": "0.1",
                              "views": Object {
                                "component": Object {
                                  "aspectHeight": 3000,
                                  "aspectWidth": 2500,
                                  "viewerUrl": "test",
                                  "zoom": false,
                                },
                                "config": Object {
                                  "viewerUrl": "test",
                                },
                                "liveConfig": Object {
                                  "viewerUrl": "test",
                                },
                                "panel": Object {
                                  "viewerUrl": "test",
                                },
                              },
                              "whitelistedConfigUrls": Array [
                                "foo",
                              ],
                              "whitelistedPanelUrls": Array [
                                "bar",
                              ],
                            }
          }
          frameId="frameid-1"
          mode="viewer"
          type="component"
/>,
        "style": Object {
          "border": "1px solid #7D55C7",
          "height": "10px",
          "left": "20px",
          "position": "absolute",
          "top": "20px",
          "width": "10px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "view",
          "extension": Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          },
          "frameId": "frameid-1",
          "mode": "viewer",
          "type": "component",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
                    Object {
                              "border": "1px solid #7D55C7",
                              "height": "10px",
                              "left": "20px",
                              "position": "absolute",
                              "top": "20px",
                              "width": "10px",
                            }
          }
>
          <ExtensionFrame
                    className="view"
                    extension={
                              Object {
                                        "authorName": "test",
                                        "channelId": "channelId",
                                        "description": "description",
                                        "iconUrl": "icon_url",
                                        "id": "id",
                                        "name": "name",
                                        "requestIdentity": false,
                                        "sku": "sku",
                                        "state": "state",
                                        "summary": "summary",
                                        "token": "token",
                                        "vendorCode": "vendorCode",
                                        "version": "0.1",
                                        "views": Object {
                                          "component": Object {
                                            "aspectHeight": 3000,
                                            "aspectWidth": 2500,
                                            "viewerUrl": "test",
                                            "zoom": false,
                                          },
                                          "config": Object {
                                            "viewerUrl": "test",
                                          },
                                          "liveConfig": Object {
                                            "viewerUrl": "test",
                                          },
                                          "panel": Object {
                                            "viewerUrl": "test",
                                          },
                                        },
                                        "whitelistedConfigUrls": Array [
                                          "foo",
                                        ],
                                        "whitelistedPanelUrls": Array [
                                          "bar",
                                        ],
                                      }
                    }
                    frameId="frameid-1"
                    mode="viewer"
                    type="component"
          />
</div>,
        "className": "view component-view",
        "style": Object {
          "height": "600px",
          "width": "800px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ExtensionFrame
            className="view"
            extension={
                        Object {
                                    "authorName": "test",
                                    "channelId": "channelId",
                                    "description": "description",
                                    "iconUrl": "icon_url",
                                    "id": "id",
                                    "name": "name",
                                    "requestIdentity": false,
                                    "sku": "sku",
                                    "state": "state",
                                    "summary": "summary",
                                    "token": "token",
                                    "vendorCode": "vendorCode",
                                    "version": "0.1",
                                    "views": Object {
                                      "component": Object {
                                        "aspectHeight": 3000,
                                        "aspectWidth": 2500,
                                        "viewerUrl": "test",
                                        "zoom": false,
                                      },
                                      "config": Object {
                                        "viewerUrl": "test",
                                      },
                                      "liveConfig": Object {
                                        "viewerUrl": "test",
                                      },
                                      "panel": Object {
                                        "viewerUrl": "test",
                                      },
                                    },
                                    "whitelistedConfigUrls": Array [
                                      "foo",
                                    ],
                                    "whitelistedPanelUrls": Array [
                                      "bar",
                                    ],
                                  }
            }
            frameId="frameid-1"
            mode="viewer"
            type="component"
/>,
          "style": Object {
            "border": "1px solid #7D55C7",
            "height": "10px",
            "left": "20px",
            "position": "absolute",
            "top": "20px",
            "width": "10px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "view",
            "extension": Object {
              "authorName": "test",
              "channelId": "channelId",
              "description": "description",
              "iconUrl": "icon_url",
              "id": "id",
              "name": "name",
              "requestIdentity": false,
              "sku": "sku",
              "state": "state",
              "summary": "summary",
              "token": "token",
              "vendorCode": "vendorCode",
              "version": "0.1",
              "views": Object {
                "component": Object {
                  "aspectHeight": 3000,
                  "aspectWidth": 2500,
                  "viewerUrl": "test",
                  "zoom": false,
                },
                "config": Object {
                  "viewerUrl": "test",
                },
                "liveConfig": Object {
                  "viewerUrl": "test",
                },
                "panel": Object {
                  "viewerUrl": "test",
                },
              },
              "whitelistedConfigUrls": Array [
                "foo",
              ],
              "whitelistedPanelUrls": Array [
                "bar",
              ],
            },
            "frameId": "frameid-1",
            "mode": "viewer",
            "type": "component",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionComponentView /> renders correctly when zoom is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionComponentView
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 2000,
                "aspectWidth": 3000,
                "zoom": true,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "height": "300px",
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    id="1"
    overlaySize={
        Object {
            "height": 600,
            "width": 800,
          }
    }
    position={
        Object {
            "x": 10,
            "y": 10,
          }
    }
    role="Broadcaster"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
                Object {
                        "height": "0.009765625px",
                        "transform": "scale(1024)",
                        "transformOrigin": "0 0",
                        "width": "0.009765625px",
                      }
        }
>
        <ExtensionFrame
                className="view"
                extension={
                        Object {
                                "authorName": "test",
                                "channelId": "channelId",
                                "description": "description",
                                "iconUrl": "icon_url",
                                "id": "id",
                                "name": "name",
                                "requestIdentity": false,
                                "sku": "sku",
                                "state": "state",
                                "summary": "summary",
                                "token": "token",
                                "vendorCode": "vendorCode",
                                "version": "0.1",
                                "views": Object {
                                  "component": Object {
                                    "aspectHeight": 2000,
                                    "aspectWidth": 3000,
                                    "zoom": true,
                                  },
                                  "config": Object {
                                    "viewerUrl": "test",
                                  },
                                  "liveConfig": Object {
                                    "viewerUrl": "test",
                                  },
                                  "panel": Object {
                                    "height": "300px",
                                    "viewerUrl": "test",
                                  },
                                },
                                "whitelistedConfigUrls": Array [
                                  "foo",
                                ],
                                "whitelistedPanelUrls": Array [
                                  "bar",
                                ],
                              }
                }
                frameId="frameid-1"
                mode="viewer"
                type="component"
        />
</div>,
      "className": "view component-view",
      "style": Object {
        "height": "600px",
        "width": "800px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ExtensionFrame
          className="view"
          extension={
                    Object {
                              "authorName": "test",
                              "channelId": "channelId",
                              "description": "description",
                              "iconUrl": "icon_url",
                              "id": "id",
                              "name": "name",
                              "requestIdentity": false,
                              "sku": "sku",
                              "state": "state",
                              "summary": "summary",
                              "token": "token",
                              "vendorCode": "vendorCode",
                              "version": "0.1",
                              "views": Object {
                                "component": Object {
                                  "aspectHeight": 2000,
                                  "aspectWidth": 3000,
                                  "zoom": true,
                                },
                                "config": Object {
                                  "viewerUrl": "test",
                                },
                                "liveConfig": Object {
                                  "viewerUrl": "test",
                                },
                                "panel": Object {
                                  "height": "300px",
                                  "viewerUrl": "test",
                                },
                              },
                              "whitelistedConfigUrls": Array [
                                "foo",
                              ],
                              "whitelistedPanelUrls": Array [
                                "bar",
                              ],
                            }
          }
          frameId="frameid-1"
          mode="viewer"
          type="component"
/>,
        "style": Object {
          "height": "0.009765625px",
          "transform": "scale(1024)",
          "transformOrigin": "0 0",
          "width": "0.009765625px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "view",
          "extension": Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 2000,
                "aspectWidth": 3000,
                "zoom": true,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "height": "300px",
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          },
          "frameId": "frameid-1",
          "mode": "viewer",
          "type": "component",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
                    Object {
                              "height": "0.009765625px",
                              "transform": "scale(1024)",
                              "transformOrigin": "0 0",
                              "width": "0.009765625px",
                            }
          }
>
          <ExtensionFrame
                    className="view"
                    extension={
                              Object {
                                        "authorName": "test",
                                        "channelId": "channelId",
                                        "description": "description",
                                        "iconUrl": "icon_url",
                                        "id": "id",
                                        "name": "name",
                                        "requestIdentity": false,
                                        "sku": "sku",
                                        "state": "state",
                                        "summary": "summary",
                                        "token": "token",
                                        "vendorCode": "vendorCode",
                                        "version": "0.1",
                                        "views": Object {
                                          "component": Object {
                                            "aspectHeight": 2000,
                                            "aspectWidth": 3000,
                                            "zoom": true,
                                          },
                                          "config": Object {
                                            "viewerUrl": "test",
                                          },
                                          "liveConfig": Object {
                                            "viewerUrl": "test",
                                          },
                                          "panel": Object {
                                            "height": "300px",
                                            "viewerUrl": "test",
                                          },
                                        },
                                        "whitelistedConfigUrls": Array [
                                          "foo",
                                        ],
                                        "whitelistedPanelUrls": Array [
                                          "bar",
                                        ],
                                      }
                    }
                    frameId="frameid-1"
                    mode="viewer"
                    type="component"
          />
</div>,
        "className": "view component-view",
        "style": Object {
          "height": "600px",
          "width": "800px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ExtensionFrame
            className="view"
            extension={
                        Object {
                                    "authorName": "test",
                                    "channelId": "channelId",
                                    "description": "description",
                                    "iconUrl": "icon_url",
                                    "id": "id",
                                    "name": "name",
                                    "requestIdentity": false,
                                    "sku": "sku",
                                    "state": "state",
                                    "summary": "summary",
                                    "token": "token",
                                    "vendorCode": "vendorCode",
                                    "version": "0.1",
                                    "views": Object {
                                      "component": Object {
                                        "aspectHeight": 2000,
                                        "aspectWidth": 3000,
                                        "zoom": true,
                                      },
                                      "config": Object {
                                        "viewerUrl": "test",
                                      },
                                      "liveConfig": Object {
                                        "viewerUrl": "test",
                                      },
                                      "panel": Object {
                                        "height": "300px",
                                        "viewerUrl": "test",
                                      },
                                    },
                                    "whitelistedConfigUrls": Array [
                                      "foo",
                                    ],
                                    "whitelistedPanelUrls": Array [
                                      "bar",
                                    ],
                                  }
            }
            frameId="frameid-1"
            mode="viewer"
            type="component"
/>,
          "style": Object {
            "height": "0.009765625px",
            "transform": "scale(1024)",
            "transformOrigin": "0 0",
            "width": "0.009765625px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "view",
            "extension": Object {
              "authorName": "test",
              "channelId": "channelId",
              "description": "description",
              "iconUrl": "icon_url",
              "id": "id",
              "name": "name",
              "requestIdentity": false,
              "sku": "sku",
              "state": "state",
              "summary": "summary",
              "token": "token",
              "vendorCode": "vendorCode",
              "version": "0.1",
              "views": Object {
                "component": Object {
                  "aspectHeight": 2000,
                  "aspectWidth": 3000,
                  "zoom": true,
                },
                "config": Object {
                  "viewerUrl": "test",
                },
                "liveConfig": Object {
                  "viewerUrl": "test",
                },
                "panel": Object {
                  "height": "300px",
                  "viewerUrl": "test",
                },
              },
              "whitelistedConfigUrls": Array [
                "foo",
              ],
              "whitelistedPanelUrls": Array [
                "bar",
              ],
            },
            "frameId": "frameid-1",
            "mode": "viewer",
            "type": "component",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
